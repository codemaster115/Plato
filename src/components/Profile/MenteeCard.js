import React from 'react';
import {
  Card,
  CardContent,
  InputLabel,
  MenuItem,
  FormControl,
  Select,
  TextField  
 } from '@material-ui/core';
import styled from 'styled-components';
import { connect } from 'react-redux';

import Button from './Button';

const TextIndex = styled.div`     
    box-sizing: border-box;    
    height: 25px;
    width: 411.418px;        
    overflow: visible;
    font-family: Lato;
    font-size: 14px;
    color: rgb(112, 112, 112);
    line-height: 1.4;
    padding-top: 3.5px;        
`;

const MentorLeft = styled.div`     
    display: grid;        
    grid-template-rows: 37px 53px 120px 55px;
`;

const SharpType = styled.div`     
    display: grid;    
    grid-gap: 5px;
    grid-template-columns: 75px 250px 75px;
`;

const StyledSelect = styled(Select)`     
    margin: 0px !important;
`;

const StyledTextField = styled(TextField)`     
    width: 300px;
`;

const StyledMultiTextField = styled(TextField)`     
    width: 300px;    
`;

const MentorRight = styled.div`     
    display: grid;        
    grid-template-rows: 37px 53px 120px;
`;

const Icons = styled.div`    
`;

const MentorIcon = styled.button`        
    box-sizing: border-box;    
    height: 20px;
    width: 17.9084px;    
    text-align: center;
    cursor: pointer;
    background: none;
    border: none;
    color: rgba(69, 75, 128, 0.7);
    font-size: 17.9084px;
    border-radius: 4px;
    float: right;
    margin-right: 10px;
    &:hover {
      color: rgb(69, 75, 128);
    }
`;

const AddMentor = styled.div`
    padding-top: 23px;
`;

const Submit = styled.div`
    text-align: right;
    color: red;
    float: right;
    padding-top: 70px;
`;

const SubButton = styled(Button)`
    padding: 8px 26px;
    background-color: rgb(199, 199, 199);
    margin-top: 5px;
`;

const MentorButton = styled(Button)`
    padding: 8px 13px;    
`;

const MentorCard = styled(CardContent)`     
    display: grid;    
    grid-gap: 10px;
    grid-template-columns: 50% 50%;
    height: 300px;
`;

class ButtonComponent extends React.Component {
  
  render() {
    return (
             <Card>
                <MentorCard>
                  <MentorLeft>
                    <SharpType>
                      <TextIndex>Share type</TextIndex>
                      <FormControl>
                        <InputLabel htmlFor="sharp-types"></InputLabel>
                        <StyledSelect
                           value=""
                          // onChange={handleChange}
                          // inputProps={{
                          //   name: 'type',
                          //   id: 'sharp-types',
                          // }}
                        >
                          <MenuItem value="">
                            <em>None</em>
                          </MenuItem>
                          <MenuItem value={1}>Spontaneous chat on Slack</MenuItem>
                          <MenuItem value={2}>CS</MenuItem>
                          <MenuItem value={3}>During an intial call</MenuItem>
                        </StyledSelect>
                      </FormControl>
                      <TextIndex>
                        <strong>
                            Temporary
                        </strong>
                      </TextIndex>
                    </SharpType>                        
                    <StyledTextField
                      id="standard-dense"
                      label="Type here..."                          
                      margin="dense"
                    />     
                    <StyledMultiTextField
                      id="standard-dense"
                      label="Type here..."                          
                      margin="dense"
                      multiline
                      rows={5}                          
                    />         
                    <StyledTextField
                      id="standard-dense"
                      label="Type here..."                          
                      margin="dense"
                    />                 
                  </MentorLeft>
                  <MentorRight>
                    <Icons>
                      <MentorIcon className="fa fa-trash-o inner-element bubble-element clickable-element"/>
                      <MentorIcon className="fa fa-info inner-element bubble-element clickable-element"/>                          
                    </Icons>
                    <AddMentor>
                      <MentorButton>Add Mentor</MentorButton>
                    </AddMentor>
                    <Submit>
                      <div>Missing CC!</div>                          
                      <SubButton>
                        Submit
                      </SubButton>
                    </Submit>
                  </MentorRight>
                </MentorCard>
              </Card>   
    );
  }
}

export default ButtonComponent;
