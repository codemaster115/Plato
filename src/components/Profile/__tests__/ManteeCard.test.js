import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MenteeCard from '../MenteeCard';

test('Compare snapshots', () => {
  const Component = renderer.create(<MenteeCard />);

  const Tree = Component.toJSON();
  expect(Tree).toMatchSnapshot();
});
test('Renders without crashing', () => {
  mount(<MenteeCard />);
});