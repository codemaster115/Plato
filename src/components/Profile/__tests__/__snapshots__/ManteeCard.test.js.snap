// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare snapshots 1`] = `
.c3 {
  box-sizing: border-box;
  height: 25px;
  width: 411.418px;
  overflow: visible;
  font-family: Lato;
  font-size: 14px;
  color: rgb(112,112,112);
  line-height: 1.4;
  padding-top: 3.5px;
}

.c1 {
  display: grid;
  grid-template-rows: 37px 53px 120px 55px;
}

.c2 {
  display: grid;
  grid-gap: 5px;
  grid-template-columns: 75px 250px 75px;
}

.c4 {
  margin: 0px !important;
}

.c5 {
  width: 300px;
}

.c6 {
  width: 300px;
}

.c7 {
  display: grid;
  grid-template-rows: 37px 53px 120px;
}

.c8 {
  box-sizing: border-box;
  height: 20px;
  width: 17.9084px;
  text-align: center;
  cursor: pointer;
  background: none;
  border: none;
  color: rgba(69,75,128,0.7);
  font-size: 17.9084px;
  border-radius: 4px;
  float: right;
  margin-right: 10px;
}

.c8:hover {
  color: rgb(69,75,128);
}

.c9 {
  padding-top: 23px;
}

.c11 {
  text-align: right;
  color: red;
  float: right;
  padding-top: 70px;
}

.c12 {
  box-sizing: border-box;
  height: 30px;
  width: 94px;
  background: none rgb(85,90,191);
  border-radius: 4px;
  box-shadow: rgb(238,241,244) 0px 2px 7px 0px;
  cursor: pointer;
  color: white;
  float: right;
  font-family: "Source Sans Pro";
  font-size: 14px;
  font-weight: 400;
  padding: 8px 11px;
  padding: 8px 26px;
  background-color: rgb(199,199,199);
  margin-top: 5px;
}

.c12:hover {
  opacity: 0.9;
}

.c10 {
  box-sizing: border-box;
  height: 30px;
  width: 94px;
  background: none rgb(85,90,191);
  border-radius: 4px;
  box-shadow: rgb(238,241,244) 0px 2px 7px 0px;
  cursor: pointer;
  color: white;
  float: right;
  font-family: "Source Sans Pro";
  font-size: 14px;
  font-weight: 400;
  padding: 8px 11px;
  padding: 8px 13px;
}

.c10:hover {
  opacity: 0.9;
}

.c0 {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 50% 50%;
  height: 300px;
}

<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
>
  <div
    className="MuiCardContent-root-29 c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
        >
          Share type
        </div>
        <div
          className="MuiFormControl-root-30"
        >
          <label
            className="MuiFormLabel-root-45 MuiInputLabel-root-34 MuiInputLabel-formControl-39 MuiInputLabel-animated-42"
            data-shrink={false}
            htmlFor="sharp-types"
          />
          <div
            className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 c4"
            onClick={[Function]}
          >
            <div
              className="MuiSelect-root-52"
            >
              <div
                aria-haspopup="true"
                aria-owns={undefined}
                aria-pressed="false"
                className="MuiSelect-select-53 MuiSelect-selectMenu-56 MuiInputBase-input-82 MuiInput-input-67"
                id="select-"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                aria-describedby={undefined}
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id={undefined}
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-89 MuiSelect-icon-58"
                color={undefined}
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="c3"
        >
          <strong>
            Temporary
          </strong>
        </div>
      </div>
      <div
        className="MuiFormControl-root-30 MuiFormControl-marginDense-32 c5"
      >
        <label
          className="MuiFormLabel-root-45 MuiInputLabel-root-34 MuiInputLabel-formControl-39 MuiInputLabel-animated-42 MuiInputLabel-marginDense-40"
          data-shrink={false}
          htmlFor="standard-dense"
        >
          Type here...
        </label>
        <div
          className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-marginDense-79"
          onClick={[Function]}
        >
          <input
            aria-describedby={undefined}
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputMarginDense-83 MuiInput-inputMarginDense-68"
            defaultValue={undefined}
            disabled={false}
            id="standard-dense"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={undefined}
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-30 MuiFormControl-marginDense-32 c6"
      >
        <label
          className="MuiFormLabel-root-45 MuiInputLabel-root-34 MuiInputLabel-formControl-39 MuiInputLabel-animated-42 MuiInputLabel-marginDense-40"
          data-shrink={false}
          htmlFor="standard-dense"
        >
          Type here...
        </label>
        <div
          className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-marginDense-79 MuiInputBase-multiline-80 MuiInput-multiline-65"
          onClick={[Function]}
        >
          <textarea
            aria-describedby={undefined}
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputMultiline-84 MuiInput-inputMultiline-69 MuiInputBase-inputMarginDense-83 MuiInput-inputMarginDense-68"
            defaultValue={undefined}
            disabled={false}
            id="standard-dense"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={5}
            value={undefined}
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-30 MuiFormControl-marginDense-32 c5"
      >
        <label
          className="MuiFormLabel-root-45 MuiInputLabel-root-34 MuiInputLabel-formControl-39 MuiInputLabel-animated-42 MuiInputLabel-marginDense-40"
          data-shrink={false}
          htmlFor="standard-dense"
        >
          Type here...
        </label>
        <div
          className="MuiInputBase-root-72 MuiInput-root-59 MuiInput-underline-63 MuiInputBase-formControl-73 MuiInput-formControl-60 MuiInputBase-marginDense-79"
          onClick={[Function]}
        >
          <input
            aria-describedby={undefined}
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInputBase-input-82 MuiInput-input-67 MuiInputBase-inputMarginDense-83 MuiInput-inputMarginDense-68"
            defaultValue={undefined}
            disabled={false}
            id="standard-dense"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={undefined}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className=""
      >
        <button
          className="fa fa-trash-o inner-element bubble-element clickable-element c8"
        />
        <button
          className="fa fa-info inner-element bubble-element clickable-element c8"
        />
      </div>
      <div
        className="c9"
      >
        <div
          className="c10"
        >
          Add Mentor
        </div>
      </div>
      <div
        className="c11"
      >
        <div>
          Missing CC!
        </div>
        <div
          className="c12"
        >
          Submit
        </div>
      </div>
    </div>
  </div>
</div>
`;
